src/naive/period.rs

```diff
@@ -0,0 +1,279 @@
+use naive::NaiveDateTime;
+use oldtime::Duration;
+
+use std::cmp::{min, max};
+
+/// A period of time between two ISO 8601 dates/times ([NaiveDateTime](NaiveDateTime)s).
```

Comment:
fwiw I think docs links only work on nightly, and docs.rs builds on nightly, so all links could be changed from the `[..](..)` to be just `[..]`. Separately, stylistically, I slightly prefer all docs links to just be simple names with backticks: e.g. 
```markdown
[`NaiveDateTime`]
```